From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Tue, 5 Sep 2017 11:01:41 +0200
Subject: fix possible not null-terminated string

fread() might not null-terminate strings when a broken database is read.

Found by covertity
---
 src/dbaccess.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/dbaccess.c b/src/dbaccess.c
index 33b939c..ffab312 100644
--- a/src/dbaccess.c
+++ b/src/dbaccess.c
@@ -37,6 +37,8 @@ int readdb(const char *iface, const char *dirname, const int force)
 			printf("db: Database loaded for interface \"%s\"...\n", data.interface);
 		}
 	}
+	data.interface[sizeof(data.interface) - 1] = 0;
+	data.nick[sizeof(data.nick) - 1] = 0;
 
 	if (data.version == DBVERSION) {
 		if (!validatedb() && !force) {
@@ -89,6 +91,8 @@ int readdb(const char *iface, const char *dirname, const int force)
 					printf("db: Backup database loaded for interface \"%s\"...\n", data.interface);
 				}
 			}
+			data.interface[sizeof(data.interface) - 1] = 0;
+			data.nick[sizeof(data.nick) - 1] = 0;
 
 			if (data.version == DBVERSION) {
 				if (!validatedb()) {
